feature-libs/asm/components/customer-emulation/customer-emulation.component.ts

```diff
@@ -1,20 +1,40 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
+import {
+  Component,
+  EventEmitter,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { OccAsmAdapter } from '@spartacus/asm/occ';
+import { ActiveCartFacade } from '@spartacus/cart/base/root';
 import { User, UserService } from '@spartacus/core';
+import { ICON_TYPE } from '@spartacus/storefront';
 import { Observable, Subscription } from 'rxjs';
 import { AsmComponentService } from '../services/asm-component.service';
-
 @Component({
   selector: 'cx-customer-emulation',
   templateUrl: './customer-emulation.component.html',
+  providers: [OccAsmAdapter],
 })
 export class CustomerEmulationComponent implements OnInit, OnDestroy {
   customer: User;
+  cartId: FormControl = new FormControl();
+  iconTypes = ICON_TYPE;
   isCustomerEmulationSessionInProgress$: Observable<boolean>;
+  cartIdExists: boolean;
+  showAssignCartSuccess = false;
+  showAssignCartError = false;
   protected subscription = new Subscription();
 
+  @Output()
+  submitEvent = new EventEmitter<{ customerId?: string }>();
+
   constructor(
     protected asmComponentService: AsmComponentService,
-    protected userService: UserService
+    protected userService: UserService,
+    protected occAsmAdapter: OccAsmAdapter,
```

Comment:
A component can not depend directly on an OCC adapter and make http call itself.
Whatever logic required to query the backend should be encapsulated by a facade service